src/Microsoft.DotNet.Build.Tasks/VersionTools/UpdateToRemoteDependencies.cs

```diff
@@ -0,0 +1,128 @@
+
+        protected override void TraceListenedExecute()
+        {
+            using (GitHubClient client = CreateClient(allowAnonymous: true))
+            {
+                DependencyUpdateResults updateResults = UpdateToRemote(client);
+
+                Log.LogMessage(
+                    MessageImportance.Low,
+                    $"Suggested commit message: '{updateResults.GetSuggestedCommitMessage()}'");
+            }
+        }
+
+        protected GitHubClient CreateClient(bool allowAnonymous)
+        {
+            GitHubAuth gitHubAuth = null;
+            if (string.IsNullOrEmpty(GitHubAuthToken))
+            {
+                if (allowAnonymous)
+                {
+                    Log.LogMessage(
+                        MessageImportance.Low,
+                        $"No value for '{nameof(GitHubAuthToken)}'. " +
+                        "Accessing GitHub API anonymously.");
+                }
+                else
+                {
+                    throw new ArgumentException(
+                        $"Required argument '{nameof(GitHubAuthToken)}' is null or empty.");
+                }
+            }
+            else
+            {
+                gitHubAuth = new GitHubAuth(GitHubAuthToken, GitHubUser, GitHubEmail);
+            }
+
+            return new GitHubClient(gitHubAuth);
+        }
+
+        protected DependencyUpdateResults UpdateToRemote(GitHubClient client)
+        {
+            // Use the commit hash of the remote dotnet/versions repo master branch.
+            string versionsCommitHash = client
+                .GetReferenceAsync(new GitHubProject("versions", "dotnet"), "heads/master")
+                .Result.Object.Sha;
+
+            DependencyUpdateResults updateResults = DependencyUpdateUtils.Update(
+                CreateUpdaters().ToArray(),
+                CreateDependencyInfos(true, versionsCommitHash).ToArray());
+
+            // Update CurrentRef and CurrentBranch for each applicable build info used.
+            if (!string.IsNullOrEmpty(CurrentRefXmlPath))
+            {
+                foreach (BuildInfo info in updateResults.UsedInfos
+                    .OfType<BuildDependencyInfo>()
+                    .Select(i => i.BuildInfo)
+                    .Distinct())
+                {
+                    ITaskItem infoItem = FindBuildDependencyInfoItem(info.Name);
+
+                    if (!string.IsNullOrEmpty(infoItem.GetMetadata(CurrentRefMetadataName)))
+                    {
+                        UpdateProperty(
+                            CurrentRefXmlPath,
+                            $"{info.Name}{CurrentRefMetadataName}",
+                            versionsCommitHash);
+                    }
+                }
+            }
+
+            return updateResults;
+        }
+
+        private ITaskItem FindBuildDependencyInfoItem(string name)
+        {
+            return DependencyInfo.SingleOrDefault(item =>
+                item.ItemSpec == name &&
```

Comment:
Should these compares be case-insensitive?

Reply:
The `BuildInfo`'s name was originally set from the `DependencyInfo`'s `ItemSpec` in `CreateBuildInfo`, so there's no reason for the case to differ.

That said, there's no good reason to do a search here to "rediscover" the connection. Really, the item and the `DependencyInfo` should be stored together.

What comes to mind to fix that is subclassing `BuildDependencyInfo` in `Microsoft.DotNet.Build.Tasks` to tack on the `ITaskItem` that created the info. (Subclassing to avoid referencing MSBuild libs in VersionTools.) I'll work on changing to this.